Sources/IntegerUtilities/Rounding.swift

```diff
@@ -0,0 +1,65 @@
+//===--- Rounding.swift ---------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift Numerics open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift Numerics project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+/// A rule that defines how to select one of the two representable results
+/// closest to a given value.
+public enum RoundingRule {
+  /// Produces the closest representable value that is less than or equal
+  /// to the value being rounded.
+  ///
+  /// This is the default rounding mode for integer shifts, including the
+  /// shift operators defined in the standard library.
+  case down
```

Comment:
What do you think about additionally providing something like:

```swift
static var `default`: Self { .down }
```

Reply:
I think the main concern would be that we might add an operation someday that wants to use a different default. round-to-nearest multiple comes to mind as a possibility.

Reply:
Ah, so default depends on operation (so either current design or name it `defaultShifting`, etc.). What about a "fastest" or "I don't care" computed var, would that also be dependent on the operation?

Reply:
"don't care" here would just be `>>`, and for divide it would be "/". Expectation is that people reach for these when they _do_ care.